test/test_criteria_evaluator.rb

```diff
@@ -329,28 +332,30 @@ def test_not_in_seg
   def test_multiple_conditions
     segment_key = 'segment_key'
 
-    segment = Prefab::Segment.new(criteria: [
-                                    Prefab::Criterion.new(
-                                      operator: Prefab::Criterion::CriterionOperator::PROP_ENDS_WITH_ONE_OF,
-                                      value_to_match: string_list(['prefab.cloud', 'gmail.com']),
-                                      property_name: 'email'
-                                    ),
-
-                                    Prefab::Criterion.new(
-                                      operator: Prefab::Criterion::CriterionOperator::PROP_IS_ONE_OF,
-                                      value_to_match: Prefab::ConfigValue.new(bool: true),
-                                      property_name: 'admin'
-                                    )
-                                  ])
-
     segment_config = Prefab::Config.new(
       config_type: Prefab::ConfigType::SEGMENT,
       key: segment_key,
       rows: [
         Prefab::ConfigRow.new(
           values: [
             Prefab::ConditionalValue.new(
-              value: Prefab::ConfigValue.new(segment: segment)
+              value: Prefab::ConfigValue.new(bool: true),
+              criteria: [
+                Prefab::Criterion.new(
+                  operator: Prefab::Criterion::CriterionOperator::PROP_ENDS_WITH_ONE_OF,
+                  value_to_match: string_list(['prefab.cloud', 'gmail.com']),
+                  property_name: 'email'
+                ),
+
+                Prefab::Criterion.new(
```

Comment:
I think, for this to test the new `OR` logic, these two criteria need to be in separate `ConditionalValue` structs.

I believe that the assertion below 
```
assert_equal DEFAULT_VALUE, evaluator.evaluate({ email: 'example@prefab.cloud' }).string
```
should, by the new desired logic, return the `DESIRED_VALUE`, since one of the segment criteria (`:PROP_ENDS_WITH_ONE_OF`) is true, even if the other criterion is false.

The `in_segment?` logic in the evaluator, unless I'm missing something, hasn't changed to allow for `any?` instead of `all?` criteria, so, as we can see by the tests passing, I don't think we're implementing the logic modeled in
![Screenshot 2023-01-24 at 11 43 53 AM](https://user-images.githubusercontent.com/28326/214472658-3592620a-0a6a-45fb-b06f-ed13f4d8fb89.jpg)